

```bash
ln -s /output/genomic/plant/Actinidia/chinensis/CK51F3_01/Genome/Assembly/PS1/1.69.0/AllChromosomes/WithoutNullChromosome/Fasta/Red5_PS1.1.69.0.fsa /workspace/hrpjxt/SNPcallingHR/ref
```


```bash

```


```bash
module avail
```

    
    ------------------------ /usr/share/Modules/modulefiles ------------------------
    dot         module-git  module-info modules     null        use.own
    
    ------------------------- /home/hrpjxt/privatemodules --------------------------
    null
    
    ------------- /software/OSutils/modules/share/Modules/modulefiles --------------
    1SynChro/1.15(default)
    A5-miseq/20160825(default)
    abyss/2.0.2(default)
    adapterremoval/2.2.2(default)
    AfterQC/0.9.1
    AfterQC/0.9.5(default)
    allpathslg/44837
    allpathslg/44913
    allpathslg/49722
    allpathslg/50191
    allpathslg/52488(default)
    amos/3.1.0(default)
    angsd/0.913
    angsd/0.917-116-g5d087b2(default)
    angsd/0.921
    angsd-wrapper/7224db2(default)
    anytag/2.5.2(default)
    apsim/7.9-r4047
    apsim/7.9-r4047-docker
    apsim/7.9-r4176(default)
    apsimx/2189-docker
    apsimx/2253(default)
    art/MountRainier(default)
    aspera-cli/3.7.2.354.010c3b8(default)
    assemblathon_stats/14dfdab(default)
    assemblathon_stats/2011_10_13
    augustus/3.2.2(default)
    augustus/3.3
    bambam/1.4(default)
    bamtools/2.3.0
    bamtools/2.4.0(default)
    bats/0.4.0(default)
    BBMap/37.28
    BBMap/37.53
    BBMap/37.93(default)
    bc/1.07(default)
    bcftools/1.3.1
    bcftools/1.6(default)
    bcgTree/1.0.8(default)
    beagle/4.0
    beagle/4.1(default)
    beast/2.4.7(default)
    BEAST/1.8.4(default)
    BEAST2/2.4.7(default)
    bedtools/2.20.1
    bedtools/2.21.0
    bedtools/2.27.1(default)
    bfc-submitted/v1(default)
    biobambam/0.0.191(default)
    biobambam2/2.0.44(default)
    bismark/0.19.1(default)
    blasr/5.2.badc092(default)
    blast/2.2.26(default)
    BLAT/36(default)
    blobtools/1.0(default)
    boost/1_61_0(default)
    boost/1_61_0-dev
    boost/1_65_1
    boost/1_65_1-dev
    bowtie/1.0.0(default)
    bowtie2/2.2.5
    bowtie2/2.2.9(default)
    BRAKER/1.11
    BRAKER/1.9
    BRAKER/2
    BRAKER/2.0.1
    BRAKER/2.0.4
    BRAKER/2.0.5
    BRAKER/2.1.0(default)
    BSseeker2/2.1.5(default)
    BUSCO/v1.2
    BUSCO/v2.0
    BUSCO/v3.0.2(default)
    bwa/0.7.12
    bwa/0.7.15
    bwa/0.7.17(default)
    canu/1.3
    canu/1.5
    canu/1.6
    canu/1.7(default)
    cd-hit/4.6.1(default)
    cdo/1.9.4(default)
    cegma/v2.4.010312
    cegma/v2.5(default)
    checkm/1.0.6(default)
    chromonomer/1.07(default)
    chromosomer/0.1.3(default)
    cndsrc/2010.10.11
    cndsrc/2017.11.30(default)
    cnvnator/0.3.3(default)
    CodingQuarry/2.0(default)
    COGNATE/v1.0(default)
    colordiff/1.0.18(default)
    conda/4.5.1(default)
    cookiecutter/1.0.0(default)
    crisp/c6cd86e(default)
    csvkit/1.0.2(default)
    cufflinks/2.2.1(default)
    cutadapt/1.10
    cutadapt/1.15(default)
    data-registry/1.0(default)
    data-registry/1.0-dev
    data-registry/1.0-test
    Disco/1.1-beta(default)
    docker-compose/1.11.2(default)
    docker-compose/1.7.1
    DWGSIM/0561546(default)
    ea-utils/1.04.807(default)
    ea-utils/1.1.2-484
    ea-utils/1.1.2-537
    ea-utils/1.1.2-806
    EffectorP/1.0(default)
    eg-all/91(default)
    eg-web-bacteria/91(default)
    eg-web-common/91(default)
    eg-web-fungi/91(default)
    eg-web-metazoa/91(default)
    eg-web-plants/91(default)
    eg-web-protists/91(default)
    eg-web-search/91(default)
    eg-web-testsuite/91(default)
    ehive/2.4(default)
    emboss/6.4.0
    emboss/6.5.7
    emboss/6.6.0(default)
    EMBOSS/6.6.0(default)
    ensembl/69
    ensembl/91
    ensembl/92(default)
    ensembl-analysis/69
    ensembl-analysis/91
    ensembl-analysis/92(default)
    ensembl-api/69
    ensembl-api/91
    ensembl-api/92(default)
    ensembl-compara/69
    ensembl-compara/91
    ensembl-compara/92(default)
    ensembl-funcgen/69
    ensembl-funcgen/91
    ensembl-funcgen/92(default)
    ensembl-genebuild/69
    ensembl-genebuild/91
    ensembl-genebuild/92(default)
    ensemblgenomes-api/91(default)
    ensembl-git-tools/d55068a(default)
    ensembl-hive/69
    ensembl-hive/91
    ensembl-hive/92(default)
    ensembl-io/69
    ensembl-io/91
    ensembl-io/92(default)
    ensembl-killlist/69
    ensembl-killlist/91
    ensembl-killlist/92(default)
    ensembl-metadata/69
    ensembl-metadata/91
    ensembl-metadata/92(default)
    ensembl-notebook/latest(default)
    ensembl-orm/69
    ensembl-orm/91
    ensembl-orm/92(default)
    ensembl-pipeline/69
    ensembl-pipeline/91
    ensembl-pipeline/92(default)
    ensembl-production/69
    ensembl-production/91
    ensembl-production/92(default)
    ensembl-taxonomy/69
    ensembl-taxonomy/91
    ensembl-taxonomy/92(default)
    ensembl-tools/69
    ensembl-tools/91
    ensembl-tools/92(default)
    ensembl-variation/69
    ensembl-variation/91
    ensembl-variation/92(default)
    ensembl-vep/69
    ensembl-vep/91
    ensembl-vep/92(default)
    e-PCR/2.3.9(default)
    ericscript/0.5.5(default)
    e-utilities/5.40(default)
    EVidenceModeler/1.1.1(default)
    evigene/02-11-2017
    evigene/18-01-2018(default)
    evigene/18-09-2013
    exonerate/2.4.0(default)
    FALCON/1.7.4(default)
    fasta/35.4.12
    fasta/36.3.8g(default)
    fast-gbs/2016.12.14
    fast-gbs/2017.01.12(default)
    fastp/0.12.5(default)
    FastQC/0.11.1
    FastQC/0.11.2
    FastQC/0.11.7(default)
    fastq_screen/v0.11.1(default)
    fastq_screen/v0.4.4
    fastq_screen/v0.5.2
    FastTree/2.1.10(default)
    fastx_toolkit/0.0.13(default)
    FGMP/1.0(default)
    FLASH/1.2.11(default)
    freebayes/1.1.0(default)
    freebayes/v1.0.2-16-gd466dde
    freetype/2.9(default)
    GapFiller/v1.10(default)
    GARM/0.7.4(default)
    gatk/1.0
    gatk/3.7
    gatk/3.8.0(default)
    Gblocks/0.91b(default)
    GEM/20130406-045632(default)
    GeMoMa/1.4.2(default)
    geneid/v1.4.4(default)
    GeneMark/4.21
    GeneMark/4.33(default)
    genomethreader/1.7.0(default)
    gffread/0.9.6(default)
    git/2.11.0
    git/2.12.2
    git/2.14.1(default)
    git/2.7.0
    git/2.9.0
    git-annex/6.20160613-g35dbe35(default)
    gitflow/15aab26(default)
    git-lfs/2.3.0(default)
    globus-cli/1.2.0
    globus-cli/1.2.2
    globus-cli/1.6.0(default)
    gmap/2016-06-09
    gmap/2017-06-20
    gmap/2017-09-30
    gmap/2018-03-01
    gmap/2018-03-25
    gmap/2018-05-11
    gmap/2018-05-30(default)
    gmato/v1.2(default)
    GMcloser/1.6.2(default)
    go/1.9(default)
    gromacs/2018(default)
    hagfish/05d08d1(default)
    hagfish/1_May_2013
    hapcompass/0.8.2(default)
    HapCUT2/0121162(default)
    Haplosim/ae652a2
    Haplosim/bd39c6f(default)
    HapTree/1.0(default)
    hdf5/1.8.17(default)
    hicup/0.5.9(default)
    hisat2/2.1.0(default)
    hmmer/3.0
    hmmer/3.1b1
    hmmer/3.1b2(default)
    HTSeq/0.6.1
    HTSeq/0.7.2
    HTSeq/0.9.1(default)
    htslib/1.3.1
    htslib/1.7(default)
    hub/2.2.3(default)
    idba/a1bafe6(default)
    IGV/2.3.93(default)
    infernal/1.1.2(default)
    JAGS/4.0.0
    JAGS/4.2.0(default)
    java/1.7.0
    java/1.8.0(default)
    jcvi/0.7.3(default)
    jcvi/0.7.7
    jellyfish/1.1.10
    jellyfish/2.1.3(default)
    kat/2.3.1(default)
    kentUtils/302.1(default)
    khmer/2.0(default)
    KMC/3.0.0(default)
    kraken/0.10.5
    kraken/1.0
    kraken/1.1(default)
    Krona/2.7(default)
    LACHESIS/20150929(default)
    last/770(default)
    lastz/1.02.00(default)
    LDhat/81596e2(default)
    links/v1.8.5(default)
    LocHap/2.0(default)
    luarocks/2.4.1(default)
    lumpy-sv/0.2.13(default)
    MACS2/2.1.1(default)
    mafft/7.307(default)
    mapcomp/1.0(default)
    MAPGD/0.4.26
    MAPGD/0.5(default)
    mason/0.1.2(default)
    MaSuRCA/3.2.2
    MaSuRCA/3.2.3
    MaSuRCA/3.2.4(default)
    mauve/2015.02.13(default)
    mcl/14-137(default)
    mcscan/0.8(default)
    MECAT/3898797(default)
    megahit/1.1.1(default)
    meme/4.11.2(default)
    Meraculous/2.2.4(default)
    Metaassembler/1.5(default)
    metabat/0.32.4(default)
    metaphlan2/2.6.0(default)
    mikado/1.1(default)
    miniasm/0.2(default)
    miniasm/55cf018
    miniconda2/4.3.13(default)
    minimap2/2.9(default)
    mira/4.0.2(default)
    miRanda/3.3a(default)
    msms/3.2rc_163(default)
    MultiQC/1.0
    MultiQC/1.2
    MultiQC/1.3
    MultiQC/1.5(default)
    MUMmer/3.23(default)
    muscle/3.8.1551
    muscle/4.0.170(default)
    ncbi-blast/2.2.25
    ncbi-blast/2.2.30
    ncbi-blast/2.6.0(default)
    NetLogo/6.0.4(default)
    newbler/2.5.3
    newbler/2.7
    newbler/2.9(default)
    nextclip/v1.3.1(default)
    NextGenMap/v0.5.5(default)
    NPStat/v1(default)
    nseg/1.0.0(default)
    NSEG/2018-05-30(default)
    NxTrim/0.4.2(default)
    NxTrim/v0.4.1-0f17575
    oases/0.2.09(default)
    obitools/1.2.11(default)
    openmpi/1.8.8(default)
    OPERA-LG/v2.0.6(default)
    orthomcl/2.0.9(default)
    pandaseq/2.10(default)
    pandoc/1.19.2(default)
    PASA/2.0.2
    PASA/2.1.0
    PASA/2.2.0
    PASA/2.3.3(default)
    patman/1.2.2(default)
    PBSIM/308622d(default)
    PBSuite/15.8.24(default)
    PBSuite/17.05.09-dbrowneup
    pear/0.9.10(default)
    PedigreeSim/v2.0(default)
    PendigreeSim/v2.0(default)
    perl/5.10.1
    perl/5.10.1-thread-multi
    perl/5.12.5
    perl/5.12.5-thread-multi
    perl/5.14.4
    perl/5.14.4-thread-multi
    perl/5.16.3
    perl/5.16.3-thread-multi
    perl/5.18.4
    perl/5.20.3
    perl/5.20.3-thread-multi
    perl/5.22.2
    perl/5.22.2-thread-multi
    perl/5.24.0
    perl/5.24.0-thread-multi
    perl/5.24.1
    perl/5.24.1-thread-multi
    perl/5.26.0(default)
    perl/5.26.0-thread-multi
    perl/5.8.9
    perl/5.8.9-thread-multi
    perl/bio-extras
    perl/bio-extras-5.24.0-legacy
    perl/core
    perl/ensembl
    perl/jupyterhub
    perlbrew/0.76(default)
    pfr-python2/2.7.13(default)
    pfr-python3/3.6.1
    pfr-python3/3.6.5(default)
    phylip/3.696(default)
    phylobayes/4.1c(default)
    PhyloSift/1.0.1(default)
    picard-tools/1.112
    picard-tools/1.79
    picard-tools/2.10.1
    picard-tools/2.18.7(default)
    picard-tools/2.2.4
    picard-tools/2.9.4
    pilon/1.18
    pilon/1.20(default)
    platanus/1.2.4(default)
    platanus_trim/1.0.7(default)
    Platypus/0.8.1(default)
    plink/1.07(default)
    popoolation2/1201(default)
    porechop/0.2.3(default)
    portcullis/1.1.0
    portcullis/1.1.1(default)
    pplacer/v1.1.alpha17-6-g5cecf99(default)
    primer3/2.2.3
    primer3/2.3.5(default)
    prinseq-lite/0.20.4(default)
    Prodigal/2.6.3(default)
    prokka/1.11(default)
    Python/3.5.1(default)
    QIIME/1.9.1(default)
    qiime2/2017.10
    qiime2/2017.7(default)
    qiime2/2017.9
    qiime2/2018.2
    qiime2/2018.6
    qualimap/v2.2.1(default)
    quast/4.2(default)
    R/3.3.0
    R/3.3.1
    R/3.3.3
    R/3.4.0
    R/3.4.1
    R/3.4.2
    R/3.4.3(default)
    R/3.4.4
    racon/1.3.0(default)
    RAILS/v1.2(default)
    RANBOW/1
    RANBOW/1-temp(default)
    RECON/1.08(default)
    REPCLASS/1.0.1(default)
    RepeatMasker-open/3-3-0
    RepeatMasker-open/4-0-3
    RepeatMasker-open/4-0-5(default)
    RepeatModeler/1.0.8(default)
    RepeatModeler-open/1.0.11(default)
    RepeatScout/1.0.5(default)
    rnacocktail/0.2.1(default)
    RSEM/1.3.0(default)
    rtg-tools/3.7.1-eb13bbb(default)
    ruby/2.1.9
    ruby/2.2.7
    ruby/2.3.4
    ruby/2.4.1(default)
    sabre/1.000(default)
    Salmon/0.8.2
    Salmon/0.9.1(default)
    samtools/0.1.18
    samtools/0.1.19
    samtools/1.1
    samtools/1.2
    samtools/1.3.1
    samtools/1.7(default)
    ScanIndel/1.2dev(default)
    SDhap/56d7678(default)
    seqkit/0.7.0(default)
    seqtk/1.2(default)
    sga/0.10.10-2
    sga/0.10.13(default)
    sga/0.9.19
    sga/0.9.35
    SGSGeneLoss/0.1(default)
    singularity/2.4.2
    singularity/2.5.1(default)
    smalt/0.7.6(default)
    snap/1.0beta.18(default)
    snapr/0.16alpha.51(default)
    SNPhylo/20140701
    SNPhylo/20160204(default)
    SOAPdenovo/1.05
    SOAPdenovo/2.04(default)
    SOAPdenovo-Trans/1.03(default)
    SolexaQA/3.1.5(default)
    sortmerna/2.1(default)
    SPAdes/3.10.1
    SPAdes/3.11.1
    SPAdes/3.12.0(default)
    SPAdes/3.7.1
    sratoolkit/2.8.2.1(default)
    SSPACE/2.0(default)
    stacks/1.40
    stacks/1.40_withbam
    stacks/2.0(default)
    standard-RAxML/8.2.11(default)
    standard-RAxML/8.2.9
    STAR/2.4.1d
    STAR/2.4.2a
    STAR/2.5.2a
    STAR/2.5.2b
    STAR/2.5.3a(default)
    STAR-Fusion/0.8.0(default)
    stringtie/1.3.3b.Linux_x86_64(default)
    subread/1.4.5
    subread/1.5.3(default)
    supernova/1.2.2(default)
    SV2/1.4.3.2(default)
    svtools/0.3.2(default)
    system/0.3(default)
    tabix/0.2.6(default)
    TargQC/1.4.4(default)
    tassel/3.0
    tassel/4.0
    tassel/5.0
    tassel/5.2.28(default)
    taxmaps/0.2.1(default)
    texlive/20151117(default)
    tophat/2.0.13
    tophat/2.1.1(default)
    totalview/2017.2.11(default)
    TransDecoder/3.0.1
    TransDecoder/5.0.1(default)
    transrate/1.0.3(default)
    trf/4.07(default)
    trim_galore/0.4.1
    trim_galore/0.4.3(default)
    Trimmomatic/0.36(default)
    trinityrnaseq/2.0.6
    trinityrnaseq/2.2.0
    trinityrnaseq/2.3.2
    trinityrnaseq/2.4.0
    trinityrnaseq/2.5.1
    trinityrnaseq/2.6.5(default)
    trinotate/3.0.2(default)
    tRNAscan-SE/1.3.1(default)
    ucsc-pslcdnafilter/324(default)
    unixODBC/2.3.0(default)
    vcflib/50a3229
    vcflib/586c5ae
    vcflib/73b7f13(default)
    vcftools/0.1.12
    vcftools/0.1.14(default)
    velvet/1.2.10(default)
    VersioningService/69
    VersioningService/91
    VersioningService/92(default)
    vg/1.7.0(default)
    ViennaRNA/2.2.2(default)
    vsearch/2.3.2(default)
    wgs/8.3rc2(default)
    whatshap/0.14.1-py36_0(default)
    wise/2.4.1(default)
    wkhtmlox/0.12.3(default)



```bash
module load samtools/1.7
module load bcftools/1.6
module load bamtools/2.4.0
module load bwa/0.7.17
```


```bash
bsub -e bwa.err -o bwa.out bwa index /workspace/hrpjxt/SNPcallingHR/ref/Red5_PS1.1.69.0.fsa 
```

    Job <376509> is submitted to default queue <normal>.



```bash
bjobs
```

    No unfinished job found





```bash
bsub -e sam.err -o sam.out samtools faidx /workspace/hrpjxt/SNPcallingHR/ref/Red5_PS1.1.69.0.fsa
```

    Job <390223> is submitted to default queue <normal>.



```bash
mkdir reads
```


```bash
ln -s /workspace/cfphxd/bioinf_Russel_GBS/011.GBSQC/ /workspace/hrpjxt/SNPcallingHR/reads
```


```bash
mkdir concareads
```

# BWA using bwasw (5 fastq.gz files aligned)


```bash
bsub -e bwa.err -o bwa.out bwa bwasw -t 4 /workspace/hrpjxt/SNPcallingHR/ref/Red5_PS1.1.69.0.fsa  /workspace/hrpjxt/SNPcallingHR/reads/011.GBSQC/P3_HR_C9NAEANXX_NoIndex_L002_R1_trimmed.fq.gz -f P3.sam
```

    Job <387247> is submitted to default queue <normal>.


# converting SAMs to BAM file


```bash
bsub -e sam.err -o sam.out samtools view -b -S -o P1.bam P1.sam
```

    Job <388249> is submitted to default queue <normal>.



```bash
bsub -e sam12.err -o sam12.out samtools view -b -S -o P1-2.bam P1-2.sam
```

    Job <388250> is submitted to default queue <normal>.



```bash
bsub -e sam2.err -o sam2.out samtools view -b -S -o P2.bam P2.sam
```

    Job <388251> is submitted to default queue <normal>.



```bash
bsub -e sam21.err -o sam21.out samtools view -b -S -o P2-1.bam P2-1.sam
```

    Job <388252> is submitted to default queue <normal>.



```bash
bsub -e sam3.err -o sam3.out samtools view -b -S -o P3.bam P3.sam
```

    Job <388253> is submitted to default queue <normal>.



```bash
bjobs
```

    No unfinished job found





```bash
samtools view P1.bam | head
```

    D00626:189:C9NREANXX:5:2301:3230:2000	16	LG25	2215567	170	96M4S	*	0	0	AGGACACATATTGACATCTCCAGTCAAATGGTCTGATACACACTGATAACAAAAGACATGTTCGCAAATGGTAATGACTGCATCTTCAGGTGGATCATGN	GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGEGGGGGGGGGGGCGGGGEGGGGGGGGGGGEGBAA=#	AS:i:92	XS:i:0	XF:i:3	XE:i:2	NM:i:1
    D00626:189:C9NREANXX:5:2301:4086:2000	0	LG26	3952076	164	9S90M	*	0	0	NCACCACGTGATCCAACCAGAGGAATAATTGGAACTATGGTATCAAACTTCTTAATAGCATTTCCTATTAGAAATGAATTTTCTAGCATTTGACTCCGT	#==ABFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGBG	AS:i:86	XS:i:0	XF:i:3	XE:i:2	NM:i:1
    D00626:189:C9NREANXX:5:2301:4623:2000	16	LG12	9904765	10	62M7S	*	0	0	GATCCAACGGAGCTCTCAGGCCTTATTTGATTTACATCCAATTAAATTTCTAATTTTGGATCAGGCGAN	GGFGGGGGGGGGGGGGGGGEGGGFGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGBBA=#	AS:i:50	XS:i:0	XF:i:1	XE:i:1	NM:i:3
    D00626:189:C9NREANXX:5:2301:4989:1999	0	ps1sf630	100027	172	6S94M	*	0	0	NGTCCTGATCCTGTTTCACCTACTTTATACATTCGATCAGGTCCTATTCTTCTACCCTTTCCCGGATCAGTTGCAATCTATCTTTCTTCTTCGGATTGTC	#=AABFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGBFGGGGG	AS:i:94	XS:i:0	XF:i:3	XE:i:3	NM:i:0
    D00626:189:C9NREANXX:5:2301:5177:1999	16	LG3	987682	169	91M9S	*	0	0	CCTCTCTCTCGCTCTCCTACTTTCCCGTTAAATGCAATACCTCACTGTATCCAGCCAAATACCTCCACCAACATAACTATGCGCAAGGATCAGTGTTCTN	GGFGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGF=GGGGGGGGGGGGGGGGGGGGGGGGGGEGGGDBA<=#	AS:i:91	XS:i:0	XF:i:3	XE:i:2	NM:i:0
    D00626:189:C9NREANXX:5:2301:5123:1999	16	LG9	10388610	0	93M7S	*	0	0	GAATCAATAGAGAGCCTTTTCTTCTGTATGAATCGATATTATTACATTCCAATTCCTTCCCGATACCTCCCAAGGAAAATATCGAATTGGATCGATGAAN	GGGGGGGGGGGGGEFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGGGGGECGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGBB<<#	AS:i:93	XS:i:93	XF:i:2	XE:i:2	NM:i:0
    D00626:189:C9NREANXX:5:2301:5565:1999	4	*	0	0	*	*	0	0	NTGCGCCAAATGCTTACTCAAGCTCAAACGGCTGGTCAGTATTTTACCAATGACCAAATCAAAGAAATGACTCGCAAGGTTAGTGCTGAGGTTGACTT	#==ABGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG
    D00626:189:C9NREANXX:5:2301:6441:2000	4	*	0	0	*	*	0	0	NCACCAGTAGATCCCGTTCGAACTTCGAAGTTAAGCGTGCATAGACAGGTGCAATGTCAGGATGGGTGACCCAGCGGGAAGTCCCCTTTTTACACCCTTT	#3<:A00E@FECGGGGGGGGFGDGGGGGGDFG1EFGGGGGGGGGGGGGGGCGGFGGGGGEGEGE@>@FFGFBGGGGGGGGGFGGGGGGGEGGGEGGGGGG
    D00626:189:C9NREANXX:5:2301:6909:1999	16	LG25	16055484	30	93M7S	*	0	0	CGATTTTTTACACTTGTAGGTCGATAGGTCCATATGATTTGAAAAGGTGAACCAACATACCTTCGCGATTATATTGGTACAAATGTATGGATCGGTCAGN	GGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGDGDGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGBA=<#	AS:i:73	XS:i:0	XF:i:3	XE:i:1	NM:i:5
    D00626:189:C9NREANXX:5:2301:7245:2000	0	LG11	2281320	0	7S93M	*	0	0	NTCGCCTGATCCTTTTCCTCGTTCGAGAATCTCCTCCTTTCTTCCACTCCGTCCCGAAGAGTAACTAGGACCAATTTAGTCACGTTTTCATGTTCCAATT	#<=<ADFCCGFGDCGG>1FC=C@FBBFGGGGCFGC@F1=<FGGGFE>F>FGBFG<>FBGDBC@GCEDFGGG>FB@BC@F1CGGB;0CBEFCE0@C;DD08	AS:i:93	XS:i:93	XF:i:1	XE:i:0	NM:i:0
    samtools view: writing to standard output failed: Broken pipe
    samtools view: error closing standard output: -1


# sorting BAMs


```bash
bsub samtools sort P1.bam -o P1.sorted.bam
```

    Job <388421> is submitted to default queue <normal>.



```bash
bsub samtools sort P1-2.bam -o P1-2.sorted.bam
```

    Job <388422> is submitted to default queue <normal>.



```bash
bsub samtools sort P2.bam -o P2.sorted.bam
```

    Job <388423> is submitted to default queue <normal>.



```bash
bsub samtools sort P2-1.bam -o P2-1.sorted.bam
```

    Job <388424> is submitted to default queue <normal>.



```bash
bsub samtools sort P3.bam -o P3.sorted.bam
```

    Job <388425> is submitted to default queue <normal>.


# index sorted BAMs


```bash
bsub samtools index P1.sorted.bam
```

    Job <389745> is submitted to default queue <normal>.



```bash
bsub samtools index P1-2.sorted.bam
```

    Job <389746> is submitted to default queue <normal>.



```bash
bsub samtools index P2.sorted.bam
```

    Job <389747> is submitted to default queue <normal>.



```bash
bsub samtools index P2-1.sorted.bam
```

    Job <389748> is submitted to default queue <normal>.



```bash
bsub samtools index P3.sorted.bam
```

    Job <389749> is submitted to default queue <normal>.


# indexing genome again


```bash
bsub -e sam.err -o sam.out samtools faidx /workspace/hrpjxt/SNPcallingHR/ref/Red5_PS1.1.69.0.fsa
```

    Job <390225> is submitted to default queue <normal>.


# mpile up to VCF


```bash
bsub -e bcf.err samtools mpileup -g -f /workspace/hrpjxt/SNPcallingHR/ref/Red5_PS1.1.69.0.fsa P1.sorted.bam -o P1.bcf
```

    Job <390240> is submitted to default queue <normal>.



```bash
bjobs
```

    JOBID   USER    STAT  QUEUE      FROM_HOST   EXEC_HOST   JOB_NAME   SUBMIT_TIME
    390240  hrpjxt  RUN   normal     aklppr31    wkoppb38    *-o P1.bcf Aug 25 01:36



```bash
bkill 390239
```

    Job <390239>: Job has already finished



